src/SickSafetyscannersRos.cpp

```diff
@@ -934,5 +943,80 @@ std::string SickSafetyscannersRos::getDateString(uint32_t days_since_1972, uint3
   return retval;
 }
 
+void SickSafetyscannersRos::scanWatchdogTimerCallback(const ros::TimerEvent& event)
+{
+  // If the timeout has expired, try to reconnect to the lidar
+  std::lock_guard<std::mutex> lock(m_watchdog_mutex);
+  auto elapsed_time = event.current_real - std::max(m_configured_stamp, m_scan_stamp);
+  if (m_initialised && elapsed_time > m_scan_timeout)
+  {
+    ROS_WARN_STREAM(
+      "No scan sector messages have been received in the last " << std::setprecision(3) << elapsed_time.toSec()
+                                                                << " seconds (since " << m_scan_stamp
+                                                                << "). Resetting the lidar.");
+    ROS_WARN_STREAM("Scan sector watchdog found an issue. Trying to reconnect to lidar.");
+    triggerReconfigure();
```

Comment:
The reconfigure is occurring while the `m_watchdog_mutex` is locked. I'm not sure if that is needed or desired.
* `m_initialised` is an atomic, so that doesn't need the mutex
* `m_configure_timer` is only accessed in the constructor and other callbacks. So unless this node uses a multi-threaded spinner, this should not need the mutex either.
* Which leaves just `m_device` to worry about. `m_device` looks to be used in other threads, but does not _seem_ to be guarded. Is `m_device` thread-safe?

Reply:
I believe that the `m_device` is only used by the spinner. Spinner creates it and destroys it (after the initial construction). The calls done with it are also done by the spinner, including reconfigure. This watchdog timer is only required to get stamps. I will limit the scope.